apps/charterafrica/src/payload/collections/Tools.js

```diff
@@ -0,0 +1,209 @@
+import {
+  TOOL_COLLECTION,
+  ORGANIZATION_COLLECTION,
+  PEOPLE_COLLECTION,
+} from "../../lib/tools/models";
+import slug from "../fields/slug";
+import source from "../fields/toolSourceField";
+import updatedAt from "../fields/updatedAt";
+
+const Tools = {
+  slug: TOOL_COLLECTION,
+  admin: {
+    useAsTitle: "externalId",
+  },
+  access: {
+    read: () => true,
+  },
+  labels: {
+    singular: {
+      en: "Tool",
+      fr: "Outil",
+      pt: "Ferramenta",
+    },
+    plural: {
+      en: "Tools",
+      fr: "Outils",
+      pt: "Ferramentas",
+    },
+  },
+  fields: [
+    source(),
+    {
+      name: "externalId",
+      type: "text",
+      required: true,
+      label: { en: "External ID", fr: "ID externe", pt: "ID externo" },
+      admin: {
+        readOnly: true,
+      },
+    },
+    slug({ fieldToUse: "name" }),
+    {
+      name: "name",
+      type: "text",
+      label: {
+        en: "Tool Name",
+        fr: "Nom de l'outil",
+        pt: "Nome da ferramenta",
+      },
+      admin: {
+        readOnly: true,
+      },
+    },
+    {
+      name: "description",
+      type: "textarea",
+      admin: {
+        readOnly: true,
+      },
+      label: {
+        en: "Tool Description",
+        fr: "Description de l'outil",
+        pt: "Descrição da ferramenta",
+      },
+    },
+    {
+      name: "link",
+      type: "text",
+      admin: {
+        readOnly: true,
+      },
+      label: {
+        en: "Tool Link",
+        fr: "Lien de l'outil",
+        pt: "Link da ferramenta",
+      },
+    },
+    {
+      name: "location",
```

Comment:
What is a tool's location? Is it like a website? e.g. for PesaYetu, that will be pesayetu.pesacheck.org

Reply:
Topic, description, location fields are based on the data filled on airtable(Tools).

These are fields that needed some cleaning.
Based on [this doc](https://docs.google.com/spreadsheets/d/1R_rfrjThEdItbXESEO0O37j758voMKxxznWVV4WyfYo/edit#gid=45126557), what we have on topic, description and location for [Commons.Africa](https://github.com/CodeForAfrica/academy.AFRICA) is not what has been filled in the sheet.

Reply:
Cool @koechkevin but this is where critical thinking should kick in:

1. We're talking about digital tools, what does physical location actually mean here? e.g. we're building charterAFRICA, what does location (whether it's a country e.g. Kenya or continent like Africa) actually mean? I kinda get location makes sense for Organization and People.
2. Tools and Organisations (I don't know about people) already contain about description. Are we just overriding them with our own Tool Description? I'm thinking for tools that we don't own, someone may get offended by our description of the tool especially if they've taken time to craft their own description.

Reply:
This discussion @koechkevin

Reply:
For the tool, we have these fields

   **externalId** Data from sheet(as Tool Github)
    **avatarUrl**: data?.openGraphImageUrl ?? null (Entirely from github API)
    **name** Data from sheet
    **description** Data from sheet
    **link**- From Sheet
    **location** - From Sheet
    **subject: topic** - From Sheet,
    **languagesTechSkills** - From Github,
    **lastCommit**: data?.defaultBranchRef - From Github,
    **stars**: data?.stargazers?.totalCount- From Github,
    **views**: data?.watchers?.totalCount - From Github,
    **forks**: data?.forks?.totalCount - From Github,
    **people** - From Github,
    **organisation** - From Github,
    **source**: "github" Internal
    **externalUpdatedAt**: new Date(data.updatedAt),
    **deletedAt:** null Track  Deleted

Reply:
For Organisation and people, we obtain everything from Github @kilemensi

Reply:
I get that @koechkevin . Lets consider the CKAN repo below
<img width="941" alt="Screenshot 2023-05-10 at 10 45 51" src="https://github.com/CodeForAfrica/ui/assets/1779590/5e252988-e937-4fa8-aa3f-45edad9f7a38">

You can see:
1. There is obviously repo `name`
2. Then there is `about` which is `description` of the repo/tool
3. And there is website link (which is probably the `link` in our model).

What I'm asking is why are we overriding these values that we can pull from GitHub with the ones manually entered in the Sheet? Do we know more about the tool to write our own description compared to the people who wrote the tool?

I may understand `name` since repo name is usually a slug of real name e.g. commons-ui vs COMMON UI.

Reply:
Cool @koechkevin.

As I mentioned, I get that we may need to have names manually set but for description and link, I'm not really sure.

1. Tools get updated all the time, do we have enough man-power to always make sure our manual descriptions and links are always up to date.
2. If it's about missing values, we can always check override if and only if we didn't receive them from GH.

Reply:
@kilemensi here's Sarah's response on this question. https://code4africa.slack.com/archives/C03SB274J4B/p1683709926400879?thread_ts=1683635508.979919&cid=C03SB274J4B

Reply:
`Tool Link` is not a link to our internal model.
A tool like Pesa Yetu for instance has a tool link https://pesayetu.pesacheck.org/

We cannot rely on GH for this information, it has to be manual.

Reply:
Not sure what you mean @koechkevin, isn't the link exactly the same as the one on the PesaYetu repo?
<img width="941" alt="Screenshot 2023-05-10 at 13 11 34" src="https://github.com/CodeForAfrica/ui/assets/1779590/ec6f48a1-7f71-4343-9049-a4758ab88feb">

Reply:
I have updated to get link (homepageUrl) and description from github @kilemensi

Reply:
Cool @koechkevin but also I've responded to Sarah on the conversation so please go through it and let me know if it makes sense.

Reply:
Should I mark this as resolved now @kilemensi ? The conversation makes sense upto this point.